src/VirtualMonoRepo/source-mappings.json

```diff
@@ -210,11 +215,23 @@
         },
         {
             "name": "winforms",
-            "defaultRemote": "https://github.com/dotnet/winforms"
+            "defaultRemote": "https://github.com/dotnet/winforms",
+            "include": [
+                "src/System.Windows.Forms.Design/src/Resources/colordlg.data",
+                "src/System.Windows.Forms/src/Resources/System/Windows/Forms/Design/Thumbs.db"
+            ],
+            "exclude": [
+                // Non-OSS license - https://github.com/dotnet/winforms/issues/10779
+                "https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/LICENSE.txt"
+            ]
         },
         {
             "name": "wpf",
-            "defaultRemote": "https://github.com/dotnet/wpf"
+            "defaultRemote": "https://github.com/dotnet/wpf",
+            "include": [
+                "src/Microsoft.DotNet.Wpf/src/PresentationFramework/**",
+                "src/Microsoft.DotNet.Wpf/src/Shared/Tracing/resources/**"
```

Comment:
Isn't this too broad? Any _new_ binaries added in the context of these paths will automatically be included.

Reply:
Or, I guess that's intended, right? We'll have validation to catch this.

Reply:
> Any new binaries added in the context of these paths will automatically be included.

This is true. What we want is for the current binaries to be allowed but any new binaries should be detected, so you're right that it should be narrower. I've updated the paths to me more specific in https://github.com/dotnet/installer/pull/18563/commits/08fff131d3dc64b1a107a0663e3ec6b6e8804a15.

Reply:
I do not think it makes sense to have the fine-grained lists. It creates a busy work. Many of these lists are for test assets. If a developer adds a new test that follows the established pattern in one of these repos and needs to add a new binary to support the test, it does not make sense to require them to fix the VMR as a follow up. It creates busy work and discourages good behavior (adding tests).

Reply:
@jkotas - This PR is an implementation of the plan outlined here: https://github.com/dotnet/source-build/issues/4088. Specifically:

> Allow the current set of OSS licensed binaries within the product but don't allow additional binaries by default. An explicit action would be needed to get new binaries into the VMR. A script will be added to the VMR to strip binaries for source-only/source-build scenarios.

If you'd like different requirements please respond on that issue.

Reply:
I do not think that the strict implementation of this rule like what's in this PR makes sense.

It sounds reasonable to me to have a gate that catches introduction of new major sources of binary files (e.g. a new directory with binaries). 

I do not think it makes sense to require listing every single binary file when specific repo tests have dependency on number of binary files and it is likely that new binary files are going to be introduced as new tests are added,

Top priority for the VMR is to optimize for efficiency of our engineering system and our engineers working on the .NET product.

Reply:
I think the groupings that @jkotas is proposing here is the right balance between maintainability yet preventing new types/groups of binaries from being added without explicit action/approval.